api/proto/teleport/legacy/client/proto/authservice.proto

```diff
@@ -2633,4 +2661,19 @@ service AuthService {
 
   // ListReleases returns a list of Teleport Enterprise releases.
   rpc ListReleases(ListReleasesRequest) returns (ListReleasesResponse);
+
+  // CreatePlugin creates a new plugin instance.
+  rpc CreatePlugin(CreatePluginRequest) returns (google.protobuf.Empty);
+  // GetPlugin returns a plugin instance by name.
+  rpc GetPlugin(types.ResourceWithSecretsRequest) returns (types.PluginV1);
+  // GetPlugins returns all plugin instances.
+  rpc GetPlugins(types.ResourcesWithSecretsRequest) returns (types.PluginListV1);
+  // DeletePlugin removes the specified plugin instance.
+  rpc DeletePlugin(types.ResourceRequest) returns (google.protobuf.Empty);
+  // DeleteAllPlugins removes all plugin instances.
+  rpc DeleteAllPlugins(google.protobuf.Empty) returns (google.protobuf.Empty);
+  // SetPluginCredentials sets the credentials for the given plugin.
+  rpc SetPluginCredentials(SetPluginCredentialsRequest) returns (google.protobuf.Empty);
+  // SetPluginCredentials sets the status for the given plugin.
+  rpc SetPluginStatus(SetPluginStatusRequest) returns (google.protobuf.Empty);
```

Comment:
Are we sure we don't want an admin-only (or `Admin`-only, even) `UpsertPlugin` rpc here too?

Reply:
In toying around in this API I haven't encountered a use-case for `UpsertPlugin` yet. It should be possible to add the method later on without much hassle?

Reply:
Hm, perhaps this :slightly_smiling_face: https://github.com/gravitational/teleport/pull/20609/files#diff-cb7927b8cb36bb63bc941389c91c41c50c69d81f892361c9f8ce939092e105fbR2730-R2731

Do we have any such admin-only methods currently?

Reply:
Oh yeah, if we're caching the resource we need a "blind" upsert operation to update the cache on an OpPut event.